srcpkgs/efibootmgr/files/kernel.d/efibootmgr.post-install

```diff
@@ -28,18 +28,25 @@ major_version=$(echo $PKGNAME | cut -c 6-)
 # look for previous entry for this major kernel version
 existing_entry=$(efibootmgr | grep "Void Linux with kernel ${major_version}")
 
-# get the boot order
-# this is required because when in the next step the existing entry is removed,
-# it is also removed from the order so it needs to be restored later
-bootorder=$(efibootmgr |grep "BootOrder: " |cut -c 12-)
-
 # if existing, remove it
 if [ "$existing_entry" != "" ]; then
-  /etc/kernel.d/post-remove/50-efibootmgr $PKGNAME
+	/etc/kernel.d/post-remove/50-efibootmgr $PKGNAME
 fi
 
-# create the new entry
-efibootmgr -qc $args -L "Void Linux with kernel ${major_version}" -l /vmlinuz-${VERSION} -u "${OPTIONS}"
-
-# restore the boot order
-efibootmgr -qo $bootorder
+# if using strict efi suffix
+if [ "x${USE_STRICT_EFI_SUFFIX}" = x1 ]; then
+	# rename vmlinuz
+	if [ -f /boot/vmlinuz-${VERSION} ]; then
+		mv -f /boot/vmlinuz-${VERSION} /boot/vmlinuz-${VERSION}.efi
+	fi
+	# create the new entry
+	efibootmgr -qc $args -L "Void Linux with kernel ${major_version}" -l /vmlinuz-${VERSION}.efi -u "${OPTIONS}"
```

Comment:
For the new codepath, I would expect we will have an entry for each version, instead of single entry for the whole major version. The current hook only allows 1 version for the whole major version, which is bad if the new version is broken.

Reply:
I'm not here to fix the hook, I just want to make it work on certain laptops, including mine.

Reply:
> For the new codepath, I would expect we will have an entry for each version, instead of single entry for the whole major version. The current hook only allows 1 version for the whole major version, which is bad if the new version is broken.

I see this as a feature: I switched from grub to efibootmgr because of this. Concretely: I want to be able to choose to boot `linux5.4` and have that always boot the latest 5.4, even if I have `linux5.10` installed.

One possibility is to have an entry for linux5.4 (which is replaced) and also entries for each individual kernel. Then the default boot entry can be selected as "linux5.4" but it's still possible to boot a particular kernel version.

Another choice would be to keep a rolling list of past kernels when updating, etc.